lib/tictactoe.rb

```diff
@@ -19,44 +19,44 @@ def run
       @current_player = @game.switch_player(@current_player, @players)
     end
     report_verdict
-    @ui.ask_to_play_again == "y" ? start_new_game : @ui.say_goodbye
+    ui.ask_to_play_again == "y" ? start_new_game : ui.say_goodbye
   end
 
   private
 
+  attr_accessor :ui, :game, :grid, :players, :first_player, :game_starter, :current_player
+
   def welcome_players
-    @ui.welcome
-    @ui.print_logo
+    ui.welcome
+    ui.print_logo
   end
 
   def print_grid_state
-    if @grid.initial_state?
-      @ui.print_grid(@grid)
-    else
-      @ui.print_grid(@grid)
-      @game.announce_move_made(@ui, @current_player, @players)
+    ui.print_grid(grid)
+    if !grid.initial_state?
+      game.announce_move_made(ui, current_player, players)
     end
   end
 
   def report_verdict
-    @ui.print_grid(@grid)
-    if @grid.verdict == :winner
-      @ui.declare_winner(@grid.winning_mark)
+    print_grid_state
+    if grid.verdict == :winner
+      ui.declare_winner(grid.winning_mark)
     else
-      @ui.declare_draw
+      ui.declare_draw
     end
   end
 
   def start_new_game
     reset_game
-    @current_player = @game.switch_player(@game_starter, @players)
+    @current_player = game.switch_player(game_starter, players)
     @game_starter = @current_player
```

Comment:
I think you don't need this line. Try commenting it out. All tests are still passing and the game play still works